doc/audit/security_auditing_HLD.md

```diff
@@ -0,0 +1,366 @@
+| Modules                  | `-w /sbin/insmod -p x -k modules`<br>`-w /sbin/rmmod -p x -k modules`<br>`-w /sbin/modprobe -p x -k modules` |
+| auth.log changes         | `-w /var/log/auth.log -p wa -k auth_logs` |
+| Monitor binary dirs      | `-w /bin -p wa -k bin_changes`<br>`-w /sbin -p wa -k sbin_changes`<br>`-w /usr/bin -p wa -k usr_bin_changes`<br>`-w /usr/sbin -p wa -k usr_sbin_changes` |
+| User group management    | `-a always,exit -F arch=b64 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management`<br>`-a always,exit -F arch=b32 -S setuid,setresuid,setreuid,setfsuid,setgid,setresgid,setregid,setfsgid -F key=user_group_management` |
+| File deletion            | `-a exit,always -F arch=b64 -S unlink -S unlinkat -F key=file_deletion`<br>`-a exit,always -F arch=b32 -S unlink -S unlinkat -F key=file_deletion` |
+| Log changes              | `-w /var/log -p wa -k log_changes` |
+| Docker related           | `-w /usr/bin/dockerd -p wa -k docker_daemon`<br>`-w /etc/docker/daemon.json -p wa -k docker_config`<br>`-w /lib/systemd/system/docker.service -p wa -k docker_service`<br>`-w /lib/systemd/system/docker.socket -p wa -k docker_socket`<br>`-a always,exit -F arch=b64 -S execve -F path=/usr/bin/docker -k docker_commands`<br>`-w /var/lib/docker/ -p wa -k docker_storage`<br>`-a always,exit -F arch=b64 -S setuid,setgid,bind,connect -F comm="/usr/bin/docker" -k docker_sys` |
+| Process audit            | `-a never,exit -F path=/usr/bin/docker  -F key=process_audit`<br>`-a never,exit -F path=/usr/bin/dockerd  -F key=process_audit`<br>`-a never,exit -F path=/usr/bin/containerd -F key=process_audit`<br>`-a never,exit -F path=/usr/bin/runc -F key=process_audit`<br>`-a never,exit -F path=/usr/bin/python* -F key=process_audit`<br>`-a exit,always -F arch=b64 -S execve -F key=process_audit`<br>`-a exit,always -F arch=b32 -S execve -F key=process_audit` |
+| Network activity         | `-a exit,always -F arch=b64 -S connect,accept,sendto,recvfrom -F key=network_activity`<br>`-a exit,always -F arch=b32 -S connect,sendto,recvfrom -F key=network_activity` |
+| Socket activity          | `-a always,exit -F arch=b64 -S socket -F key=socket_activity`<br>`-a always,exit -F arch=b32 -S socket -F key=socket_activity` |
+
+### 3.3 Configuration design
+#### 3.3.1 ConfigDB schema
+```
+{
+    "AUDIT": {
+      "config": {
+        "enable": "true"
+      }
+    }
+}
+```
+| Key       | Description                 |
+| --------- | --------------------------- |
+| enable    | enable or not enable the AUDIT, the default value is true  |
+
+The AUDIT config is in the redis CONFIG_DB. The redis dictionary key is AUDIT|config.
+
+| Key       | Description                 |
+| --------- | --------------------------- |
+| enabled   | The flag indicating whether the AUDIT enabled, 1 enabled, others not enabled  |
```

Comment:
Is this table in StateDB? I guess you miss some description paragraph.

Reply:
Updated, table will be in ConfigDB